controllers/Cart.controller.js

```diff
@@ -0,0 +1,150 @@
+const cartdb = require("../models/cart.model");
+const userdb = require("../models/user.model");
+
+module.exports.addToCart = async (req, res) => {
+  const { id } = req.params;
+  const { productId } = req.body;
+  try {
+    const user = await userdb.findById(id);
+    const cart = await cartdb.findById(user.cart);
+    if (cart) {
+      if (!cart.cartItems.some(({ product }) => product == productId)) {
+        cart.cartItems.push({
+          product: productId,
+          quantity: 1,
+        });
+        await cart.save();
+      }
+    } else {
+      const newCart = await cartdb.create({
+        by: id,
+      });
+      await user.update({
+        cart: newCart._id,
+      });
+      await newCart.cartItems.push({
+        product: productId,
+        quantity: 1,
+      });
```

Comment:
I think this push statement will work w/o using `await` . Correct me i am wrong !